### PR TITLE
1046 toml cleanup

### PR DESCRIPTION
**Pull request by rprospero**
This is a clean-up of some stylistic issues on the TOML code.

1) Following the suggestion from @IVlaD17, the `serialise` and `deserialise` routines have been moved into an interface class.  This class also provides a convenient namespace for utility functions

2) Move `serialise` and `deserialise` onto the British spelling

3) Ensure that `serialise` is always const

4) Use a type alias for the fairly complicated TOML type.  As an added bonus, I've tried to structure this so that we could swap out TOML for YAML and only need to edit the `base/toml.h` file.  Not that we ever want to do that.

5) Move more code into helper functions.  I think that there's a way that we can use regular overloading to escape needing so many templates, but that is still a work in progress.

Questions to think about during your review:

1) Is there a better name than TomlTable?  I assume that there has to be.
2) Is there a better place for the `Serialisable` class than `base/toml.h`.  I don't like that the class name and the file name are unrelated.

### CODE REVIEW

File: src/base/toml.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+#pragma once
+
+#include <toml11/toml.hpp>
+
+#include <map>
+#include <vector>
+
+// The type we use for the nodes of our serialisation tree
+using TomlTable = toml::basic_value<toml::discard_comments, std::map, std::vector>;
```
Comment (trisyoungs) [inline src/base/toml.h:12]: Perhaps `SerialisedData`? Slightly clunky, I admit, but it does abstract things even further away from Toml should we ever decide to change.

File: src/classes/atomtype.h [hunk 2]
```diff
@@ -4,11 +4,11 @@
 #pragma once
 
 #include "base/enumoptions.h"
+#include "base/toml.h"
```
Comment (trisyoungs) [inline src/classes/atomtype.h:7]: If we move to `SerialisedData`, the include could be `base/serialiser.h` or similar.

**Review by trisyoungs — COMMENTED** [review]
Lots of nice cleaning up here. Some suggestions for renaming made in the comments - feel free to reject!

**Comment by rprospero** [comment]
I settled on `SerialisedValue` as part of a move to using `toml::value`.  This covers the fact that it might contain different types of values (e.g. table or array).

**Comment by IVlaD17** [comment]
> I settled on `SerialisedValue` as part of a move to using `toml::value`. This covers the fact that it might contain different types of values (e.g. table or array).

That sounds appropriate.

**Pull request merged**